freetextresponse/private/view.js

```diff
@@ -4,46 +4,125 @@ function FreeTextResponseView(runtime, element) {
     var $ = window.jQuery;
     var $element = $(element);
     var buttonSubmit = $element.find('.check.Submit');
-    var textareaParent = $element.find('.student_answer').parent();
+    var textareaParent = $element.find('.textarea').parent();
     var usedAttemptsFeedback = $element.find('.action .used-attempts-feedback');
     var problemProgress = $element.find('.problem-progress');
     var submitParent = $element.find('.Submit').parent();
-    var wordCountError = $element.find('.word-count-error');
     var submissionReceivedMessage = $element.find('.submission-received');
     var url = runtime.handlerUrl(element, 'submit');
     var userInputClass = 'user-input';
+    var buttonSave = $element.find('.save');
+    var saveParent = $element.find('.save').parent();
+    var urlSave = runtime.handlerUrl(element, 'save_reponse');
+    var userAlertMessage = $element.find('.user_alert');
+    var statusIndicator = $element.find('.status');
+    var textarea = $element.find('.textarea');
+    var textareaParentParent = textareaParent.parent();
+    
+    function _setBorderColor(indicator_class) {
+        if ( 'correct' == indicator_class ) {
+            textarea.css('border', 'solid #1e9348');
+        } else if ( 'incorrect' == indicator_class ) {
+            textarea.css('border', 'solid #b20610');
+        } else {
+            textarea.css('border', 'solid #e4e4e4');
+        }
+    }
+
+    function _setIndicators(indicator_class, indicator_visibility_class) {
+        // Set Indicator icon
+        textareaParent.removeClass('correct');
+        textareaParent.removeClass('incorrect');
+        textareaParent.removeClass('unanswered');
+        textareaParent.addClass(indicator_class);
+        textareaParentParent.removeClass('correct');
+        textareaParentParent.removeClass('incorrect');
+        textareaParentParent.removeClass('unanswered');
+        textareaParentParent.addClass(indicator_class);
+        //Border Color
+        if ( 'hidden' == indicator_visibility_class ) {
+            _setBorderColor('unanswered');
+        } else {
+            _setBorderColor(indicator_class);
+        }
+    }
 
     buttonSubmit.on('click', function () {
         buttonSubmit.text('Checking...');
-        runtime.notify('submit', {
-            message: 'Submitting...',
-            state: 'start'
-        });
+        //runtime.notify('submit', {
+        //    message: 'Submitting...',
+        //    state: 'start'
+        //});
```

Comment:
@caseylitton FYI: Rather than commenting that out multiple times throughout the codebase,
it may be better served to polyfill the method. You could do something like:

```javascript
// The "or" (`||`) makes sure we don't redefine the method if it already exists.
runtime.notify = runtime.notify || function () {
    // The logging statement is optional.
    console.log('POLYFILL runtime.notify', arguments);
};
```

This way, you only make one additive change, instead of removing code in many places.
The latter can be prone to mistakes.